pkg/expression/constant_propagation.go

```diff
@@ -388,6 +388,49 @@ func (s *propOuterJoinConstSolver) setConds2ConstFalse(filterConds bool) {
 	}
 }
 
+func (s *basePropConstSolver) dealWithPossibleHybridType(col *Column, con *Constant) (*Constant, bool) {
+	if !col.GetType().Hybrid() {
+		return con, true
+	}
+	if col.GetType().GetType() == mysql.TypeEnum {
+		d, err := con.Eval(s.ctx, chunk.Row{})
+		if err != nil {
+			return nil, false
+		}
```

Comment:
I think we need to prevent this from being reused by plan cache.
```suggestion
		d, err := con.Eval(s.ctx, chunk.Row{})
		if err != nil {
			return nil, false
		}
		if expression.MaybeOverOptimized4PlanCache(s.ctx, con) {
			s.ctx.GetSessionVars().StmtCtx.SetSkipPlanCache(errors.New("TODO"))
		}
```